### DISCUSSION TITLE
Unsupported cipher or incorrect key length ( first run )

### DISCUSSION THREAD

**Discussion by spectroman** [firefly-iii/firefly-iii#9234] [Firefly III]
Hi,

As I try to run this for the first time, I am unable to get around the API_KEY thing.

I have read and tried the other issues lodged here, but they are "different" around people who already had this running or are trying different things.

I have tries to get started in K8s, but the Helm does not work (incredibly with a similar error about Encryption Key) . In order to understand better how to run it, I am trying to install in a regular Apache (LAMP)  with PHP 8.3.11.

So following the manual e other issues, there is a strict conversation on 32 char "string" key. Right.

My go to generation tool is: `openssl rand -hex 32` which failed.

Then I see it is offered: `php artisan key:generate`

And in there I get the same error that occurs with any other thing I try:

```
# php artisan key:generate

   RuntimeException 

  Unsupported cipher or incorrect key length. Supported ciphers are: aes-128-cbc, aes-256-cbc, aes-128-gcm, aes-256-gcm.

  at vendor/laravel/framework/src/Illuminate/Encryption/Encrypter.php:62
     58▕ 
     59▕         if (! static::supported($key, $cipher)) {
     60▕             $ciphers = implode(', ', array_keys(self::$supportedCiphers));
     61▕ 
  ➜  62▕             throw new RuntimeException("Unsupported cipher or incorrect key length. Supported ciphers are: {$ciphers}.");
     63▕         }
     64▕ 
     65▕         $this->key = $key;
     66▕         $this->cipher = $cipher;

      +23 vendor frames 

  24  app/Exceptions/Handler.php:208
      Illuminate\Auth\AuthManager::__call()
      +3 vendor frames 

  28  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()
```

The previous error appear if there is a 32 char generate by `openssl` with different methods, or something from `base64` (like suggested in other issue)  -- Then OK -- Lets remove the key and try to run empty and see what happens...

```
 # php artisan key:generate

   Illuminate\Encryption\MissingAppKeyException 

  No application encryption key has been specified.

  at vendor/laravel/framework/src/Illuminate/Encryption/EncryptionServiceProvider.php:83
     79▕     protected function key(array $config)
     80▕     {
     81▕         return tap($config['key'], function ($key) {
     82▕             if (empty($key)) {
  ➜  83▕                 throw new MissingAppKeyException;
     84▕             }
     85▕         });
     86▕     }
     87▕ }

      +26 vendor frames 

  27  app/Exceptions/Handler.php:208
      Illuminate\Auth\AuthManager::__call()
      +3 vendor frames 

  31  artisan:35
      Illuminate\Foundation\Console\Kernel::handle()

```

So I don't know what is really going on here... which command , what can I do , that will make API_KEY work?

Or this is not API_KEY at all? 

What is happening?

**Comment by JC5** [comment]
Mmm, don't you mean APP_KEY?

**Reply by spectroman** [reply]
Yes.
